lib/feeder_importer.rb

```diff
@@ -1,64 +1,153 @@
+#     s3://prx-up/prod/<guid>/<original-filename>
+#   - copy this cms location back to the feeder original url, so they are in sync
+#   - We need to do the same for the podcast and episode images
+#
+# For the distribution, create this right away in the cms db using the feeder episode data
+#
+# After all this is setup in cms, we can then update feeder references to cms
+#   - add the `prx_uri` and `prx_account_uri` to the podcast attributes
+#   - add the `prx_uri` to the episodes
+#   - update the episode page urls (not wordpress or libsyn or whatever anymore)
+
 require 'addressable/uri'
+require 'prx_access'
+
+# Feeder models start
+# some pidgin models for feeder db
+class FeederModel < ActiveRecord::Base
+  self.abstract_class = true
+
+  def self.status_values
+    [ :started, :created, :processing, :complete, :error, :retrying, :cancelled ]
+  end
+
+  def self.feeder_db_connection
+    {
+      adapter: 'postgresql',
+      encoding: 'unicode',
+      pool: ENV['FEEDER_DB_POOL_SIZE'] || 2,
+      user: ENV['FEEDER_DB_USER'] || 'feeder',
+      password: ENV['FEEDER_DB_PASSWORD'],
+      host: ENV['FEEDER_DB_HOST'],
+      port: ENV['FEEDER_DB_PORT'] || 5432,
+      database: ENV['FEEDER_DB_DATABASE'] || 'feeder'
+    }
+  end
+
+  establish_connection(feeder_db_connection)
+end
+
+class Podcast < FeederModel
+  has_many :episodes, -> { order('published_at desc') }
+  has_many :itunes_images, -> { order('created_at DESC') }
+  has_many :feed_images, -> { order('created_at DESC') }
+end
+
+class PodcastImage < FeederModel
+  belongs_to :podcast
+  enum status: status_values
+end
+
+class ITunesImage < PodcastImage; end
+class FeedImage < PodcastImage; end
+
+class Episode < FeederModel
+  belongs_to :podcast
+  has_many :episode_images, -> { order('created_at DESC').complete }
+  has_many :media_resources, -> { order('position ASC, created_at DESC').complete }
+  scope :published, -> { where('published_at IS NOT NULL AND published_at <= now()') }
+
+  serialize :categories, JSON
+  serialize :keywords, JSON
+
+  def item_guid
+    original_guid || "prx_#{podcast_id}_#{guid}"
+  end
+end
+
+class EpisodeImage < FeederModel
+  belongs_to :episode
+  enum status: status_values
+end
+
+class MediaResource < FeederModel
+  belongs_to :episode
+  enum status: status_values
+end
+
+class Content < MediaResource; end
+class Enclosure < MediaResource; end
+# Feeder models end
 
 class FeederImporter
-  include PRXAccess
   include Announce::Publisher
+  include PRXAccess
 
-  attr_accessor :account_id, :user_id, :feeder_podcast_url
-  attr_accessor :podcast, :series, :template, :distribution
+  attr_accessor :account_id, :user_id, :podcast_id, :set_episode_urls
+  attr_accessor :podcast, :series, :template, :distribution, :stories
+
+  def debug
+    TRUE
+  end
 
-  def initialize(account_id, user_id, feeder_podcast_url)
+  def initialize(account_id, user_id, podcast_id, set_episode_urls = false)
```

Comment:
when would set_episode_urls be true?

Reply:
when we are importing a feed from libsyn, and the episode urls are on libsyn, then we want to set them as libsyn will be retired. When a producer has set the episode urls to go to their own website, and they want to keep those urls, then we would set it to false.